contracts/PricerRule.sol

```diff
@@ -0,0 +1,408 @@
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./PriceOracleInterface.sol";
+import "./EIP20TokenInterface.sol";
+import "./TokenRules.sol";
+import "./SafeMath.sol";
+
+
+contract PricerRule is Organized {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Events */
+
+    event PriceOracleAdded(
+        address _priceOracle
+    );
+
+    event PriceOracleRemoved(
+        address _priceOracle
+    );
+
+    event AcceptanceMarginAdded(
+        bytes3 _quoteCurrencyCode,
+        uint256 acceptanceMargin
+    );
+
+    event AcceptanceMarginRemoved(
+        bytes3 _quoteCurrencyCode
+    );
+
+
+    /* Storage */
+
+    /**
+     * @dev The code of the base currency of the economy.
+     */
+    bytes3 public baseCurrencyCode;
+
+    /**
+     * @dev EIP20 token address of the economy.
+     */
+    EIP20TokenInterface public eip20Token;
+
+    /**
+     * @dev Conversion rate from the base currency to the economy token.
+     */
+    uint256 public conversionRateFromBaseCurrencyToToken;
+
+    /**
+     * @dev Conversion rate decimals from the base currency to the
+     *      economy token.
+     */
+    uint256 public conversionRateDecimalsFromBaseCurrencyToToken;
+
+    /**
+     * @dev Token rules address of the economy.
+     */
+    TokenRules private tokenRules;
+
+    /**
+     * @dev Mapping from pay-currency code to corresponding price oracle.
+     *      The mapped price oracle's base currency code should be the economy
+     *      base currency code (see baseCurrencyCode storage variable)
+     *      and the quote currency code should be the corresponding pay-currency
+     *      code.
+     */
+    mapping(bytes3 => PriceOracleInterface) private baseCurrencyPriceOracles;
+
+    /**
+     * @dev Mapping from pay-currency code to price difference accepatance
+     *      margin. During a pay operation an intended price for
+     *      pay-currency is presented, that is checked against current price of
+     *      pay-currenency wrt stored acceptance margin.
+     */
+    mapping(bytes3 => uint256) private baseCurrencyPriceAcceptanceMargins;
+
+
+    /* Special Functions */
+
+    /**
+     * @notice Constructs a new pricer object.
+     *
+     * @dev Function requires:
+     *          - The economy token address is not null.
```

Comment:
Minor comment: I propose we start saying "zero" instead of "null" when we are referring to `address(0)`.

The confusion of what null (or, really, "empty") means recently lead to a bug in a test that was only realized in upgrading to 0.5.0. I'm not certain whether we should understand them to be synonyms, but I also think there's no real value to begging the question.

Open to your thoughts!

Reply:
Oh, would be happy to have an agreement here!
What are your thoughts about: bytes32(0), bytes(0) ... ?

Reply:
I think we tend to refer to that rhetorically as "empty", like here: "The base currency code is not empty."

I think that's OK (because it seems neither inaccurate nor misleading to me).